src/utils.ml

```diff
@@ -0,0 +1,46 @@
+(* Bind operator *)
+let (>>=) x f = match x with
+  | Some x -> Some (f x)
+  | None -> None
+
+(* Elvis operator ?: *)
+let (|?) x default = match x with
+  | Some x -> x
+  | None -> default
+
+
+let concatl = List.fold_left Filename.concat ""
+let path_of_list = concatl
+
+let rec list_of_path = function
+  | "/" | "." as p -> [p]
+  | p -> (Filename.basename p) :: list_of_path (Filename.dirname p)
+
+let realpath = function
+  | p when Filename.is_relative p -> Filename.concat (Sys.getcwd ()) p
+  | p -> p
+
+let rec path_eql p p' = match (p, p') with
+  | (".", ".") | ("/", "/") -> true
+  | (".", _) | (_, ".") | ("/", _) | (_, "/") -> false
+  | (_, _) -> path_eql (Filename.dirname p) (Filename.dirname p')
+
+let rewind dir file =
+  let dir = realpath dir in
+  let rec rew = function
+    | p when path_eql p dir -> ""
+    | "." | "/" -> failwith @@ "rewind: " ^ file ^ " cannot be rewinded to dir " ^ dir
+    | p -> Filename.concat ".." @@ rew @@ Filename.dirname p
+  in
+  file |> realpath |> Filename.dirname |> rew
+
+let rec remove_prefixl l l' = match (l, l') with
+  | (l, []) | ([], l) -> l
+  | (x :: l, y :: l') when x = y -> remove_prefixl l l'
+  | (_, _) -> failwith "remove_prefixl: no list is a prefix of the other"
+
+(* Return `rest' such as `path/rest=sub'
```

Comment:
Unclear.
Use ocamldoc comments `(** *)`.

May be something like: `[pathdiff sub path] returns ...`.
Explain what happens if sub is not a prefix of the string.

The function does not have the same name.